chainstate/src/detail/ban_score.rs

```diff
@@ -0,0 +1,145 @@
+use crate::BlockError;
+
+use super::{
+    pow::error::ConsensusPoWError, spend_cache::error::StateUpdateError, CheckBlockError,
+    CheckBlockTransactionsError, ConsensusVerificationError, OrphanCheckError,
+};
+
+// TODO: use a ban_score macro in a form similar to thiserror::Error in order to define the ban score
+//       value of an error on the error enum arms instead of separately like in this file
+
+pub trait BanScore {
```

Comment:
I don't think `chainstate` is the right place for this trait, if anything it should be in P2P but I can understand how that might create a weird dependency. I'd rather see this in `common` because P2P and maybe some other subsystems also use this and I think ban score is not really a "detail of chainstate" but rather a core part of swarm management.

Reply:
As discussed on slack, this whole structure will change when we attempt to use a semi-derive macro to define the scores.